### PR TITLE
JREY: Loading Screen

### PR DESCRIPTION
**Pull request by jreynard-code**
The aim of this feature is to provide to users a "loading screen" indicating the status of all data fetching (scenarios, workspaces, datasets,...). 
When all data are fetched, the application is "READY" and then, the user is redirected to main page (i.e. Scenario view)

Refactoring of App.js:
- Remove useless debug* function
- extract signIn() function into SignUtils.js to improve readability code and perform tests easily
- Place Provider in index.js instead of App.js
Add App.test.js
Add dataError, dataLoaded, dataLoading Lottie animations
Add LoadingLine component and snapshot tests
Add translations
Add Loading view
Add dependencies:
- react-fade-in
- react-loading
- react-lottie
Add devDependencies for tests:
- @react-mock/state
- axios-mock-adapter
- jest-canvas-mock
- react-test-renderer
- redux-mock-store
Setup tests (e.g. React Testing Library) 
Fix Cypress SignIn test
update yarn.lock

### CODE REVIEW

File: src/__tests__/App.test.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+// - The above error occurred in the <ForwardRef(Modal)> component
+// - The above error occurred in the <ForwardRef(Tabs)> component
+// See thread : https://stackoverflow.com/questions/67227163/material-ui-react-test-renderer-react-router-tabs-tab-component-and-modal
+describe('App test suite with loading value to false', () => {
+  const setState = jest.fn();
+
+  beforeEach(() => {
+    useStateMock
+      .mockImplementationOnce(init => [init, setState])
+      .mockImplementationOnce(init => [init, setState])
```
Comment (csm-thu) [inline src/__tests__/App.test.js:76]: This line is duplicated, was it on purpose ?

File: src/views/Loading/Loading.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    logout();
+  }
+
+  return (authenticated && application.status !== APPLICATION_STATUS.READY
+    ? (<div style={{
+        position: 'absolute',
+        left: '50%',
+        top: '50%',
+        transform: 'translate(-50%, -50%)'
+      }} data-cy="loading-component">
```
Comment (csm-thu) [inline src/views/Loading/Loading.js:48]: Could we use className to separate CSS from the JSX code ?

**Review by csm-thu — APPROVED** [review]
Good job !

File: src/__tests__/App.test.js (hunk 1 above)
Comment (jreynard-code) [reply src/__tests__/App.test.js:76]: We use 3 times the fonction useState and I want a particular implementation only for the last one.
I wanted to use mockImplementation instead but it does not allow me to deal specifically with the last call (except if I missed something :s)

**Review by COSMO-VMI — APPROVED** [review]
LGTM

**Pull request merged**